diff --git a/meson.build b/meson.build
index 800b70f15..d46b28cb8 100644
--- a/meson.build
+++ b/meson.build
@@ -90,7 +90,14 @@ libmath = cc.find_library('m')
 libpeas = dependency('libpeas-1.0', version: '>= 1.24.0')
 libsecret = dependency('libsecret-1', version: '>= 0.11')
 libsoup = dependency('libsoup-3.0')
-libstemmer_dep = cc.find_library('stemmer')
+libstemmer_enabled = get_option('libstemmer')
+
+if libstemmer_enabled
+  libstemmer_dep = cc.find_library('stemmer', required: true)
+else
+  libstemmer_dep = dependency('', required: false)  # dummy
+endif
+
 libunwind_dep = dependency(
   'libunwind', version: '>= 1.1', required: get_option('libunwind')
 )

diff --git a/meson_options.txt b/meson_options.txt
index b8ccb3cd3..3a1f8cbf4 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -76,3 +76,8 @@ option(
   value: false,
   description: 'Add delay in session opening (for debug purpose only)'
 )
+option(
+  'libstemmer', 
+  type: 'boolean', 
+  value: true,
+       description: 'Enable support for libstemmer')
