diff --git a/schemas/org.gnome.desktop.interface.gschema.xml.in b/schemas/org.gnome.desktop.interface.gschema.xml.in
index f855144..0a0d7c9 100644
--- a/schemas/org.gnome.desktop.interface.gschema.xml.in
+++ b/schemas/org.gnome.desktop.interface.gschema.xml.in
@@ -105,7 +105,7 @@
       </description>
     </key>
     <key name="font-name" type="s">
-      <default>'Adwaita Sans 11'</default>
+      <default>'Monospace 11'</default>
       <summary>Default font</summary>
       <description>
         Name of the default font used by gtk+.
