diff --git a/meson.build b/meson.build
index e2e40fd..4c7f526 100644
--- a/meson.build
+++ b/meson.build
@@ -27,9 +27,7 @@ gee_dep = dependency ('gee-0.8')
 gtk_dep = dependency ('gtk4', version: '>= 4.6')
 adw_dep = dependency ('libadwaita-1', version: '>= 1.8.alpha')
 librsvg_dep = dependency ('librsvg-2.0', version: '>=2.32.0')
-libgnome_games_support = subproject ('libgnome-games-support',
-	    		             default_options: 'superproject_name=gnome-mines')
-libgnome_games_support_dep = libgnome_games_support.get_variable ('lggs_dependency')
+libgnome_games_support_dep = dependency ('libgnome-games-support-3', version: '>= 3.0.0')
 libmath_dep = cc.find_library('m')
 
 subdir ('po')
